src/hw5/COMP232LinkedBinarySearchTree.java

```diff
@@ -170,8 +194,37 @@ private void addNodeToSubTree(BTNode<K, V> subTreeRoot,
 	 * {@inheritDoc}
 	 */
 	public V remove(K key) {
-		// Intentionally not implemented - see homework assignment.
-		throw new UnsupportedOperationException("Not yet implemented");
+		BTNode<K, V> cur = root;
+		while (true) {
+			if (cur == null) {
+				return null;
+			} else if (cur.key == key) {
+				if (cur.parent.left == cur) {
+					cur.parent.left = null;
+				} else if (cur.parent.right == cur) {
+					cur.parent.right = null;
+				}
```

Comment:
I see what you are trying to do here, but if our tree only has the root, then this will cause an error since cur.parent is null.